tests/models/test_pyro.py

```diff
@@ -36,25 +34,30 @@ def __init__(self, in_features, out_features):
         #     dist.Normal(self.zero, self.ten).expand([out_features]).to_event(1)
         # )
 
-    @staticmethod
-    def _get_fn_args_from_batch(tensor_dict):
-        x = tensor_dict[_CONSTANTS.X_KEY]
-        y = tensor_dict[_CONSTANTS.LABELS_KEY]
-
-        return (x, y.squeeze(1)), {}
-
     def forward(self, x, y):
         sigma = pyro.sample("sigma", dist.Uniform(self.zero, self.ten))
         mean = self.linear(x).squeeze(-1)
         with pyro.plate("data", x.shape[0]):
             pyro.sample("obs", dist.Normal(mean, sigma), obs=y)
         return mean
 
-    def guide(self, x, y):
-        return self._auto_guide(x, y)
+
+class BayesianRegressionModule(PyroBaseModuleClass):
+    def __init__(self, in_features, out_features):
+
+        super().__init__()
+        self.model = BayesianRegressionPyroModel(in_features, out_features)
+        self.guide = AutoDiagonalNormal(self.model)
```

Comment:
:+1:

Reply:
@fritzo -- is there a way to load in the `AutoGuide` parameters on a fresh initialization? It seems like some data needs to be first passed through.

Reply:
We usually simply `torch.save(model, ...)` instead of `torch.save(model.state_dict(), ...)` since the former also saves pythonic data other than tensors. Would that work for you?

Reply:
Isn't the former problematic if there are API changes/refactors? Maybe it's not an issue for the autoguide specifically?

Reply:
Hmm as I understand tensors and non-tensors are equally susceptible to API changes/refactors: tensors and non-tensor data structures will both be loaded from the saved version, but code will be loaded from the new version since pickle saves classes by name rather than by code.

In practice Pyro's autoguides are very stable: there has been only [one API change](https://github.com/pyro-ppl/pyro/pull/2127) since Pyro 1.0 release, and in fact that changed the tensors as well as code. I have never had an issue with model mismatch because my models tend to be short lived. Note Pyro's [release policy](https://github.com/pyro-ppl/pyro/blob/dev/RELEASE-MANAGEMENT.md) is to not guarantee compatibility of saved models across releases, but in practice the format is very stable.

Reply:
I see, thanks for the info. We will explore options, but do you see anything problematic with taking a training step before loading the state dict? Considering every model should be wrapped in a nn.Module.

Reply:
> taking a training step before loading the state dict

Hmm that should probably work, as long as you `pyro.clear_param_store()` after taking that step.

Reply:
Thanks, I suppose we'll try this option for now.